### ISSUE TITLE
Support PHPUnit attributes for `php_unit_internal_class` and `php_unit_test_class_requires_covers`

### ISSUE THREAD

**Issue by Brainshaker95** [PHP-CS-Fixer/PHP-CS-Fixer#6795]
## Feature request

I have noticed that the two rules `php_unit_internal_class` and `php_unit_test_class_requires_covers` don't work weill together with PHPUnit attributes.

```diff
<?php

declare(strict_types=1);

namespace App\Tests;

use App\Foo\Bar;
use PHPUnit\Framework\Attributes\CoversClass;
use PHPUnit\Framework\TestCase;

/**
 * @internal
 */
#[CoversClass(Bar::class)]
+/**
+ * @internal
+ *
+ * @coversNothing
+ */
final class MyTest extends TestCase
{
    public function testSomething(): void
    {
        // ...
    }
}
```

When I run the fixer a doc block containing `@internal` and `@coversNothing` will be added underneath the class attributes.  
I propose two things:
- Add an option for both rules that configures where newly created doc blocks should be inserted (above or below attributes). If there already is a doc block for the class it should get appended to that one.
- Adjust `php_unit_test_class_requires_covers` to recognize the `CoversClass` attribute.

I have already looked at the implementations but did not really manage to work something out yet. If someone else wants to give it a try here are the places to start at:
- [PhpUnitInternalClassFixer](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/src/Fixer/PhpUnit/PhpUnitInternalClassFixer.php#L94)
- [PhpUnitTestClassRequiresCoversFixer](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/src/Fixer/PhpUnit/PhpUnitTestClassRequiresCoversFixer.php#L77)

Please let me know your thoughts or suggestions on this.  
And btw I really love working with this tool, thanks to all contributers ❤

**Comment by keradus** [comment]
Thanks for pointing it out! Indeed, PHP CS Fixer does not handle PHPUnit's attributes at all.
would you like to propose a PR to bring the support for those attribs?

**Comment by Brainshaker95** [comment]
I already did try to work it out myself, but I am not sure what the best solution would be. I am also not sure if there are other rules affected by this undesirable behavior. These should be adjusted as well.   
If we could work out a concrete plan I am more than happy to give it a go.

**Comment by Brainshaker95** [comment]
I think [this](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/src/Fixer/Phpdoc/PhpdocLineSpanFixer.php#L130) is exactly what is needed for the `AbstractPhpUnitFixer` right [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/src/Fixer/AbstractPhpUnitFixer.php#L45) or am I missing something?  
Maybe this method could be extracted out and shared between these two classes? Where would I put that?

This would inevitably change the current behavior regarding the search for existing docblocks of the following classes:
- `PhpUnitInternalClassFixer`
- `PhpUnitMethodCasingFixer`
- `PhpUnitSizeClassFixer`
- `PhpUnitTestAnnotationFixer`

I don't think that this majorly breaks something but before I go ahead and implement that I wanted to ask for additional input.

`PhpUnitTestClassRequiresCoversFixer` is not affected by this change. I think this needs a custom implementation.

**Comment by Wirone** [comment]
Not reproducible on 3.48.1-DEV, after #6795 was merged.

![image](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/f0d3e2eb-09e4-4c3c-9ee2-801473aa1fae)

If there's any other scenario that is not covered, please report new issue.
